### DISCUSSION TITLE
Rule to filter results based on view

### DISCUSSION THREAD

**Discussion by chris-metz** [pocketbase/pocketbase#2765] [Q&A] [answered]
Lets say I have the collections

- users
- roles
- organizations
- products

where each user can have many roles, a role links to an organization, and each product belongs to an organization. 

Is it possible to create a rule that represents this model so only the matching products get returned to the user?

**Comment by ganigeorgiev** [comment, marked as answer]
It depends on the type of the relations and where each relation field reside, but if we assume the following:
- `users` has multiple `relation` field "roles" pointing to many `roles` record
- `roles` has a single relation field "organization" pointing to one `organizations` record
- `products` has a single `relation` field "organization" pointing to one `organizations` record

then an example `products` rule could look like this:

```js
@request.auth.id != "" &&
@request.auth.roles.organization ?= organization 
```

The above roughly translates to:
- require the user to be authenticated
- require **at least one of** the authenticated user's role (_this is because of the `?` prefix_) to point to the same organization as the product's organization

(_If the product's "organization" field is marked as "Non empty" then you can skip the first condition_)

**Reply by chris-metz** [reply]
ah great, thanks
